[RNMobile] Adding WPMediaLibrary source to File block

etoledom opened the pull request:
<!-- Learn the overall process and best practices for pull requests at https://github.com/WordPress/gutenberg/blob/master/docs/contributors/repository-management.md#pull-requests. -->

## Description
<!-- Please describe what you have changed or added -->

`gutenberg-mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2805
`WPiOS`: https://github.com/wordpress-mobile/WordPress-iOS/pull/15291
`WPAndroid`: https://github.com/wordpress-mobile/WordPress-Android/pull/13413

This PR adds the WordPress Media Library as a media source for File block.

## How has this been tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->

- Run the example app.
- Add a file block.
- Press on `Choose a file`.
  - Check that the option `WordPress Media Library` is present.
- Select such option.
  - Check that `WordPress.zip` file has been added to the block.
  - (This is mocked on the example app).

## Screenshots <!-- if applicable -->

<img width="419" alt="Screenshot 2020-11-18 at 11 05 18" src="https://user-images.githubusercontent.com/9772967/99516174-2b17b180-298e-11eb-8718-9ece49f2dc11.png">

<kbd><img src="https://user-images.githubusercontent.com/1509205/99341534-120ce480-2858-11eb-82a5-368f40c3bf9c.png" width="320"></kbd>

## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->

etoledom committed:
Adding WPMediaLibrary source for File block

jd-alexander committed:
Implemented Android side of WPMediaLibrary source for File block

jd-alexander committed:
There's now a direct WP MediaPicker action for the ALL allowed type

etoledom commented:
@Guarani could you please take a look at the iOS side changes?
Thanks! 🙏

jd-alexander committed:
set the value of "other" for the OTHER enum.

guarani commented on packages/react-native-editor/ios/GutenbergDemo/GutenbergViewController.swift:
```diff
@@ -311,12 +313,13 @@ extension GutenbergViewController: GutenbergBridgeDataSource {
 extension Gutenberg.MediaSource {
-    static let filesApp = Gutenberg.MediaSource(id: "files-app", label: "Pick a file", types: [.image, .video, .audio, .other])
+    static let filesApp = Gutenberg.MediaSource(id: "files-app", label: "Pick a file", types: [.all])
+	static let otherApp = Gutenberg.MediaSource(id: "other-apps", label: "Other Apps", types: [.image, .video, .audio, .other])
```
```suggestion
    static let otherApp = Gutenberg.MediaSource(id: "other-apps", label: "Other Apps", types: [.image, .video, .audio, .other])
```

guarani commented on packages/block-editor/src/components/index.native.js:
```diff
@@ -26,6 +26,7 @@ export {
 	default as MediaUpload,
 	MEDIA_TYPE_IMAGE,
 	MEDIA_TYPE_VIDEO,
+	MEDIA_TYPE_ALL,
```
nitpick: `MEDIA_TYPE_ANY` seems more intuitive, but that's just my opinion.

guarani reviewed (approved):
Looking good from the iOS side (I didn't test Android), left some comments, but no blockers.

etoledom committed:
Update packages/react-native-editor/ios/GutenbergDemo/GutenbergViewController.swift

Co-authored-by: Paul Von Schrottky <paul.von.schrottky@automattic.com>

etoledom committed:
Changing MEDIA_TYPE_ALL to MEDIA_TYPE_ANY

etoledom commented:
Thank you for the review @guarani.
I have updated the PR with the requested changes.

I have also updated the Android native side (cc @jd-alexander )
These changes might break WP apps PRs, so we will need to update WPAndroid and WPiOS too with the `ALL -> ANY` change.

jd-alexander committed:
When MEDIA_TYPE_ANY is passed the default device icon is shown.

jd-alexander committed:
Added a pickerTitle for the any type that represents the File block.

jd-alexander commented:
Hi @etoledom I updated the media source picker's title and added an icon for the "Choose from device" option that displays an icon on Android. 

Android | iOS 
--------|-------
<kbd><img src="https://user-images.githubusercontent.com/1509205/100153363-003bca80-2e72-11eb-88cb-17752c236e82.png" width="320"></kbd>  |     <kbd><img src="https://user-images.githubusercontent.com/1509205/100153370-02058e00-2e72-11eb-8b8e-b0f5cb64e0dd.png" width="320"></kbd>

Here, we have two variations of the button label for using a file from the device. 

Should we be using. 
1. Choose from device
2. Pick a file

I opted for "Choose from device" because it's used for the other blocks. Let me know what you think. 

----------

Even though this solution works I noticed that the "Choose from device" option for the File block is below the WP media option but for the other blocks, it's the first option. Below is a clear example. 

  Android | iOS 
--------|-------
<kbd><img src="https://user-images.githubusercontent.com/1509205/100153961-e353c700-2e72-11eb-956f-b67d8ef69f05.png" width="320"></kbd>  |     <kbd><img src="https://user-images.githubusercontent.com/1509205/100153370-02058e00-2e72-11eb-8b8e-b0f5cb64e0dd.png" width="320"></kbd>

This a minor thing. [I looked at the code and it seems the internal sources ](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/media-upload/index.native.js#L101)eg. WP Media Library is appended to the other media options which we have provided hence the ordering. Should we modify this in any way?

etoledom committed:
Change media picker option title to "Choose from device"

etoledom commented:
> Here, we have two variations of the button label for using a file from the device.

We should use `Choose from device` as specified in the design file.
I have changed the title for the iOS Demo app.
WPiOS already shows `Choose from device` 👍

jd-alexander reviewed (approved):
The behavior works as expected based on the tests that were conducted on both platforms. LGTM 🚢
